File: Java/Biblioteca/Libro.java
import java.time.LocalDate;

public class Libro extends Pubblicazione {
    private String ISBN;
    private String autore;
    //costruttore
    public Libro(String titolo, LocalDate dataPubblicazione, LocalDate dataRestituzione, boolean disponibile, String RecapitoLettore, int numPagine, String ISBN, String autore) {
        super(titolo, dataPubblicazione, dataRestituzione, disponibile, RecapitoLettore, numPagine);
        this.ISBN = ISBN;
        this.autore = autore;
    }

    //getter e setter
    public String getISBN() {
        return ISBN;
    }

    public void setISBN(String iSBN) {
        this.ISBN = iSBN;
    }

    public String getAutore() {
        return autore;
    }

    public void setAutore(String autore) {
        this.autore = autore;
    }

    @Override
    public void setDataRestituzione() {
        this.dataRestituzione = LocalDate.now().plusDays(2);
    }

    //metodo

    public String toString() {
        return super.toString() + ", ISBN=" + ISBN + ", autore=" + autore;
    }


}

File: Python/Libro/Libro.py
class Libro:
    def __init__(self,titolo,autore,pagine):
        self.titolo = titolo
        self.autore = autore
        self.pagine = pagine
        self.disponibile = True
        
    def descrizione(self):
        return "Libro - titolo: " + self.titolo + " autore: " + self.autore + " pagine: " + str(self.pagine) + " disponibile: " + str(self.disponibile) + "."
         
    def presta(self):
        self.disponibile = False
        
    def restituisci(self):
        self.disponibile = True
        